### PR TITLE
Feature/reflectivity

### PR DESCRIPTION
**Pull request by PiotrMrozik**
This pull request introduces the concept of reflectivity to the RaytraceNode, allowing for the calculation and configuration of reflectivity values in ray-tracing operations.

### Reflectivity Feature Addition:

* Added `RGL_FIELD_REFLECTIVITY_F32` to the `rgl_field_t`.
* Introduced the function `rgl_node_raytrace_configure_reflectivity_alpha` to configure reflectivity alpha values for RaytraceNode 
* Modified GPU kernels to handle reflectivity calculations and storage.

### Testing:

* Added a new test file `reflectivityTest.cpp` to cover the new reflectivity functionality.

### CODE REVIEW

File: include/rgl/api/core.h [hunk 1]
```diff
@@ -892,6 +893,15 @@ RGL_API rgl_status_t rgl_node_raytrace_configure_beam_divergence(rgl_node_t node
  */
 RGL_API rgl_status_t rgl_node_raytrace_configure_default_intensity(rgl_node_t node, float default_intensity);
 
+/**
+ * Modifies RaytraceNode to set reflectivity alpha.
+ * Reflectivity alpha is used to calculate reflectivity of the hit point. This value is constant for every hit point.
+ * Default reflectivity alpha is set to 0.1f.
```
Comment (prybicki) [inline include/rgl/api/core.h:905]: What's the reason behind this `0.1f` choice?

File: src/gpu/nodeKernels.cu [hunk 2]
```diff
@@ -127,6 +127,9 @@ __device__ void saveReturnAsHit(const RaytraceRequestContext* ctx, int beamIdx,
 		                                       static_cast<uint8_t>(std::round(ctx->mrSamples.intensity[sampleIdx])) :
 		                                       UINT8_MAX;
 	}
+	if (ctx->reflectivityF32 != nullptr) {
+		ctx->reflectivityF32[returnPointIdx] = ctx->mrSamples.reflectivity[sampleIdx];
+	}
```
Comment (prybicki) [inline src/gpu/nodeKernels.cu]: Here you can use selected sample to compute reflectivity. This way reflectivity can be removed from multi-return samples to save memory.
This suggestion was NOT COMPILED.
```suggestion
	if (ctx->reflectivityF32 != nullptr) {
		auto distance2 = ctx->mrSamples.intensity[sampleIdx] * ctx->mrSamples.intensity[sampleIdx];
		ctx->reflectivityF32[returnPointIdx] = reflectivityAlpha * ctx->mrSamples.intensity[sampleIdx] * distance2;
	}
```

File: src/gpu/MultiReturn.hpp [hunk 3]
```diff
@@ -27,6 +27,7 @@ struct MultiReturnSamplesPointers
 	Field<IS_HIT_I32>::type* isHit;
 	Field<DISTANCE_F32>::type* distance;
 	Field<INTENSITY_F32>::type* intensity;
+	Field<REFLECTIVITY_F32>::type* reflectivity;
```
Comment (prybicki) [inline src/gpu/MultiReturn.hpp]: Remove

File: test/src/scene/reflectivityTest.cpp [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+#include <RGLFields.hpp>
+struct ReflectivityTest : public RGLTestWithParam<std::tuple<float, unsigned char>>
+{};
+
+INSTANTIATE_TEST_SUITE_P(Parametrized, ReflectivityTest,
+						 testing::Combine(testing::Values(0.0f, 0.1f, 10.0f), testing::Values(u_char(0), u_char(127), u_char(255))));
+
+TEST_P(ReflectivityTest, read_value)
+{
+	auto [alpha, value] = GetParam();
```
Comment (prybicki) [inline test/src/scene/reflectivityTest.cpp]: Please parametrize also distance.

**Review by prybicki — CHANGES_REQUESTED** [review]

File: include/rgl/api/core.h (hunk 1 above)
Comment (PiotrMrozik) [reply include/rgl/api/core.h:905]: No particular reason. There should be some default, and a value smaller than 1.0 (for now) is safer regarding "exploding values".
Happy to hear any suggestions.

File: test/src/scene/reflectivityTest.cpp [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+	EXPECT_RGL_SUCCESS(rgl_graph_get_result_data(yieldNode, INTENSITY_F32, outIntensity.data()));
+	EXPECT_RGL_SUCCESS(rgl_graph_get_result_data(yieldNode, REFLECTIVITY_F32, outReflectivity.data()));
+	EXPECT_RGL_SUCCESS(rgl_graph_get_result_data(yieldNode, DISTANCE_F32, outDistance.data()));
+
+	for (int i = 0; i < outCount; ++i) {
+		EXPECT_NEAR(((float) value), outIntensity.at(i), EPSILON_F);
+		float outDdistance = outDistance.at(i);
+		float intensity = outIntensity.at(i);
+		float reflectivityValue = alpha * outDdistance * outDdistance * intensity;
+		printf("Distance: %f, Intensity: %f, Reflectivity: %f, Alpha: %f\n", outDdistance, intensity, reflectivityValue, alpha);
```
Comment (prybicki) [inline test/src/scene/reflectivityTest.cpp]:
```suggestion
		float outDistance = outDistance.at(i);
		float intensity = outIntensity.at(i);
		float reflectivityValue = alpha * outDistance * outDistance * intensity;
```

**Review by prybicki — APPROVED** [review]

**Pull request merged**